nixos/modules/virtualisation/wsl-config.nix

```diff
@@ -0,0 +1,48 @@
+{ pkgs, lib, ... }:
+
+with lib;
+
+{
+  imports = [
+  ];
+
+  config = {
+
+    boot.isContainer = true;
+
+    # Allow root logins
+    services.openssh = {
+      enable = true;
+      permitRootLogin = "prohibit-password";
+      passwordAuthentication = mkDefault false;
+    };
+
+    # configure internationalisation properties
+    i18n = {
+      defaultLocale = "en_US.UTF-8";
+    };
+
+    # automatic gc
+    nix.gc.automatic = true;
+    nix.gc.dates = "weekly";
+    nix.gc.options = "--delete-older-than 30d";
+
+    # force getting the hostname from wsl
+    networking.hostName = mkDefault "";
+
+    environment.systemPackages = with pkgs; [
+      curl
+      dos2unix
+      git-lfs
+      gitAndTools.gitFull
+      htop
+      iotop
+      sudo
+      tmux
+      vim_configurable
+      wget
+      zip

```

Comment:
This list seems rather opinionated. Should the default packages installed in NixOS not already enough?

Reply:
Indeed they are, that's what I need to be personally productive. When this merges we should ship the default packages.